src/LibTeXPrintf.jl

```diff
@@ -1,87 +1,83 @@
-
-function __init__()
-    # this is a workaround so `stdout` can work correctly and be captured by Suppressor
-    # libtexprintf.texprintf("")
-    TEXSYMBOLS[] = ImmutableDict([
-        if length(line) == 4
-            String(line[2] * line[3]) => String(line[4])
-        else
-            String(line[2]) => String(line[3])
-        end for line in split.(split(strip(capture_out() do
-                    libtexprintf.texlistsymbols()
-                end, '\n'), '\n'))
-    ]...)
+export texstring, texprint, texprintln, texsymbols, texfonts, texsetfont!, texgetfont,
+    texsetascii, texsetunicode, textogglesupersub
+
+# set line width
+texsetlw!(lw) = libtexprintf.TEXPRINTF_LW[] = lw
+
+function _texsymbols()
+    i = 1
+    symbols = Vector{Pair{String,Char}}()
+    table = LibTeXPrintf.libtexprintf.TEXPRINTF_SYMBOLS[]
+    sym = unsafe_load(table, i)
+    while sym.name != C_NULL
+        push!(symbols, unsafe_string(sym.name) => Char(sym.unicode))
+        sym = unsafe_load(table, i)
+        i += 1
+    end
+    return symbols
 end
 
 """
-    TEXSYMBOLS::Ref{ImmutableDict{String, String}}
+    TEXSYMBOLS::ImmutableDict{String, Char}
 
 Dictionary with all the symbols that the library supports.
 """
-const TEXSYMBOLS = Ref{ImmutableDict{String,String}}()
+const TEXSYMBOLS = ImmutableDict(_texsymbols()...)
 
 """
     texsymbols()
 
 Returns an immutable dictionary with all the LaTeX symbols that the library supports.
 """
-texsymbols() = TEXSYMBOLS[]
+texsymbols() = TEXSYMBOLS
 
 """
-    TEXFONTS::Vector{String}
+    TEXFONTS::NTuple{String}
 
-Dictionary with all the symbols that the library supports.
+Tuple with all the font types that the library supports.
 """
-const TEXFONTS =
-    (
-        "mathsfbfit",
-        "mathsfbf",
-        "mathfrak",
-        "mathbfit",
-        "mathsfit",
-        "mathcal",
-        "mathscr",
-        "mathbf",
-        "mathbb",
-        "mathsf",
-        "mathtt",
-        "mathnormal",
-        "text"
-    )
+const TEXFONTS = (
+    "mathbb",
+    "mathbf",
+    "mathbfit",
+    "mathcal",
+    "mathfrak",
+    "mathnormal",
+    "mathscr",
+    "mathsf",
+    "mathsfbf",
+    "mathsfbfit",
+    "mathsfit",
+    "mathtt",
+    "text",
+)
 
 """
     texfonts()
 
-Returns a `Tuple` with all the LaTeX fonts that the library supports.
+Returns a `Vector` with all the LaTeX fonts that the library supports.
 
-See also [`texsetfont`](@ref), [`texgetfont`](@ref).
+See also [`texsetfont!`](@ref), [`texgetfont`](@ref).
 
 # Examples
 ```jldoctest
 julia> texfonts()
-("mathsfbfit", "mathsfbf", "mathfrak", "mathbfit", "mathsfit", "mathcal", "mathscr", "mathbf", "mathbb", "mathsf", "mathtt", "mathnormal", "text")
+("mathbb", "mathbf", "mathbfit", "mathcal", "mathfrak", "mathnormal", "mathscr", "mathsf", "mathsfbf", "mathsfbfit", "mathsfit", "mathtt", "text")
 ```
 """
 texfonts() = TEXFONTS
 
 """
-    texsetfont(font::String)
+    texsetfont!(font::String)
```

Comment:
Argument is over-typed; should be `AbstractString`.

Reply:
Thanks, corrected.